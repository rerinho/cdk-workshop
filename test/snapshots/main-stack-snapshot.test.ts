import * as cdk from 'aws-cdk-lib'
import { MainStack } from '@lib/stacks/main-stack'
import { Template } from 'aws-cdk-lib/assertions'

test('should match the snapshot', () => {
  const app = new cdk.App()
  const stack = new MainStack(app, 'MainStack')
  const template = Template.fromStack(stack)

  expect(template.toJSON()).toMatchSnapshot()
})
