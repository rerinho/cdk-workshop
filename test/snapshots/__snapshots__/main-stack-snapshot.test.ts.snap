// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
{
  "Outputs": {
    "HelloEndpointB03699DE": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "HelloEndpointDB6339C6",
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "HelloEndpointDeploymentStageprod3D2B4BB5",
            },
            "/",
          ],
        ],
      },
    },
    "ViewHitCounterViewerEndpointCA1B1E4B": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "ViewHitCounterViewerEndpoint5A0EF326",
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "ViewHitCounterViewerEndpointDeploymentStageprodF3901FC7",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "HelloEndpointANY3E4DEFAC": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "HelloHitCounterHitCounterHandlerDAEA7B37",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Fn::GetAtt": [
            "HelloEndpointDB6339C6",
            "RootResourceId",
          ],
        },
        "RestApiId": {
          "Ref": "HelloEndpointDB6339C6",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "HelloEndpointANYApiPermissionMainStackHelloEndpoint8F25014DANY10AE3DBB": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "HelloHitCounterHitCounterHandlerDAEA7B37",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "HelloEndpointDB6339C6",
              },
              "/",
              {
                "Ref": "HelloEndpointDeploymentStageprod3D2B4BB5",
              },
              "/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "HelloEndpointANYApiPermissionTestMainStackHelloEndpoint8F25014DANY51CCBADB": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "HelloHitCounterHitCounterHandlerDAEA7B37",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "HelloEndpointDB6339C6",
              },
              "/test-invoke-stage/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "HelloEndpointAccount5C50D287": {
      "DeletionPolicy": "Retain",
      "DependsOn": [
        "HelloEndpointDB6339C6",
      ],
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "HelloEndpointCloudWatchRole338F94E5",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "HelloEndpointCloudWatchRole338F94E5": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "HelloEndpointDB6339C6": {
      "Properties": {
        "Name": "HelloEndpoint",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "HelloEndpointDeploymentC454D4F19bbe12709a642b9d026e32c20d520e7b": {
      "DependsOn": [
        "HelloEndpointproxyANY86F07935",
        "HelloEndpointproxy8620F60B",
        "HelloEndpointANY3E4DEFAC",
      ],
      "Properties": {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": {
          "Ref": "HelloEndpointDB6339C6",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "HelloEndpointDeploymentStageprod3D2B4BB5": {
      "DependsOn": [
        "HelloEndpointAccount5C50D287",
      ],
      "Properties": {
        "DeploymentId": {
          "Ref": "HelloEndpointDeploymentC454D4F19bbe12709a642b9d026e32c20d520e7b",
        },
        "RestApiId": {
          "Ref": "HelloEndpointDB6339C6",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "HelloEndpointproxy8620F60B": {
      "Properties": {
        "ParentId": {
          "Fn::GetAtt": [
            "HelloEndpointDB6339C6",
            "RootResourceId",
          ],
        },
        "PathPart": "{proxy+}",
        "RestApiId": {
          "Ref": "HelloEndpointDB6339C6",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "HelloEndpointproxyANY86F07935": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "HelloHitCounterHitCounterHandlerDAEA7B37",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "HelloEndpointproxy8620F60B",
        },
        "RestApiId": {
          "Ref": "HelloEndpointDB6339C6",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "HelloEndpointproxyANYApiPermissionMainStackHelloEndpoint8F25014DANYproxy5A870FC2": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "HelloHitCounterHitCounterHandlerDAEA7B37",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "HelloEndpointDB6339C6",
              },
              "/",
              {
                "Ref": "HelloEndpointDeploymentStageprod3D2B4BB5",
              },
              "/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "HelloEndpointproxyANYApiPermissionTestMainStackHelloEndpoint8F25014DANYproxy505EDE11": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "HelloHitCounterHitCounterHandlerDAEA7B37",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "HelloEndpointDB6339C6",
              },
              "/test-invoke-stage/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "HelloHandler2E4FBA4D": {
      "DependsOn": [
        "HelloHandlerServiceRole11EF7C63",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "a70d29913730abc9dfe0f2ad346535d054869f0d2aa2c72f393a2f266e1ca1dc.zip",
        },
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
          },
        },
        "Handler": "index.hello",
        "Role": {
          "Fn::GetAtt": [
            "HelloHandlerServiceRole11EF7C63",
            "Arn",
          ],
        },
        "Runtime": "nodejs16.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "HelloHandlerServiceRole11EF7C63": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "HelloHitCounterHitCounterHandlerDAEA7B37": {
      "DependsOn": [
        "HelloHitCounterHitCounterHandlerServiceRoleDefaultPolicy1487A60A",
        "HelloHitCounterHitCounterHandlerServiceRoleD45002B8",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "aa3ff74f7e8cdc51277187d630ea55745dccab37aa686c8ccf95ff50d2e89fe8.zip",
        },
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "DOWNSTREAM_FUNCTION_NAME": {
              "Ref": "HelloHandler2E4FBA4D",
            },
            "HITS_TABLE_NAME": {
              "Ref": "HelloHitCounterHits7AAEBF80",
            },
          },
        },
        "Handler": "index.hitCounter",
        "Role": {
          "Fn::GetAtt": [
            "HelloHitCounterHitCounterHandlerServiceRoleD45002B8",
            "Arn",
          ],
        },
        "Runtime": "nodejs16.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "HelloHitCounterHitCounterHandlerServiceRoleD45002B8": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "HelloHitCounterHitCounterHandlerServiceRoleDefaultPolicy1487A60A": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:ConditionCheckItem",
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
                "dynamodb:DescribeTable",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "HelloHitCounterHits7AAEBF80",
                    "Arn",
                  ],
                },
                {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
            {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "HelloHandler2E4FBA4D",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "HelloHandler2E4FBA4D",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "HelloHitCounterHitCounterHandlerServiceRoleDefaultPolicy1487A60A",
        "Roles": [
          {
            "Ref": "HelloHitCounterHitCounterHandlerServiceRoleD45002B8",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "HelloHitCounterHits7AAEBF80": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AttributeDefinitions": [
          {
            "AttributeName": "path",
            "AttributeType": "S",
          },
        ],
        "KeySchema": [
          {
            "AttributeName": "path",
            "KeyType": "HASH",
          },
        ],
        "ProvisionedThroughput": {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
        "SSESpecification": {
          "SSEEnabled": true,
        },
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
    "ViewHitCounterRendered9C783E45": {
      "DependsOn": [
        "ViewHitCounterRenderedServiceRoleDefaultPolicy9ADB8C83",
        "ViewHitCounterRenderedServiceRole254DB4EA",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "d9cdd7bdd784d6248d9a68dc7b1e49029a879f67840bd9cd5a3fd41de8b64c15.zip",
        },
        "Environment": {
          "Variables": {
            "SORT_BY": "-hits",
            "TABLE_NAME": {
              "Ref": "HelloHitCounterHits7AAEBF80",
            },
            "TITLE": "Hello Hits",
          },
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "ViewHitCounterRenderedServiceRole254DB4EA",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "ViewHitCounterRenderedServiceRole254DB4EA": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ViewHitCounterRenderedServiceRoleDefaultPolicy9ADB8C83": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:ConditionCheckItem",
                "dynamodb:DescribeTable",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "HelloHitCounterHits7AAEBF80",
                    "Arn",
                  ],
                },
                {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ViewHitCounterRenderedServiceRoleDefaultPolicy9ADB8C83",
        "Roles": [
          {
            "Ref": "ViewHitCounterRenderedServiceRole254DB4EA",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "ViewHitCounterViewerEndpoint5A0EF326": {
      "Properties": {
        "Name": "ViewerEndpoint",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "ViewHitCounterViewerEndpointANY66F2285B": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "ViewHitCounterRendered9C783E45",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Fn::GetAtt": [
            "ViewHitCounterViewerEndpoint5A0EF326",
            "RootResourceId",
          ],
        },
        "RestApiId": {
          "Ref": "ViewHitCounterViewerEndpoint5A0EF326",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "ViewHitCounterViewerEndpointANYApiPermissionMainStackViewHitCounterViewerEndpoint90EA86D9ANY751EDFCB": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "ViewHitCounterRendered9C783E45",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "ViewHitCounterViewerEndpoint5A0EF326",
              },
              "/",
              {
                "Ref": "ViewHitCounterViewerEndpointDeploymentStageprodF3901FC7",
              },
              "/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ViewHitCounterViewerEndpointANYApiPermissionTestMainStackViewHitCounterViewerEndpoint90EA86D9ANY0BA5884B": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "ViewHitCounterRendered9C783E45",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "ViewHitCounterViewerEndpoint5A0EF326",
              },
              "/test-invoke-stage/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ViewHitCounterViewerEndpointAccount0B75E76A": {
      "DeletionPolicy": "Retain",
      "DependsOn": [
        "ViewHitCounterViewerEndpoint5A0EF326",
      ],
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "ViewHitCounterViewerEndpointCloudWatchRole87B94D6A",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "ViewHitCounterViewerEndpointCloudWatchRole87B94D6A": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "ViewHitCounterViewerEndpointDeployment1CE7C576aaeb52ee3dd5ec3ab243f366d5eae8a9": {
      "DependsOn": [
        "ViewHitCounterViewerEndpointproxyANYFF4B8F5B",
        "ViewHitCounterViewerEndpointproxy2F4C239F",
        "ViewHitCounterViewerEndpointANY66F2285B",
      ],
      "Properties": {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": {
          "Ref": "ViewHitCounterViewerEndpoint5A0EF326",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "ViewHitCounterViewerEndpointDeploymentStageprodF3901FC7": {
      "DependsOn": [
        "ViewHitCounterViewerEndpointAccount0B75E76A",
      ],
      "Properties": {
        "DeploymentId": {
          "Ref": "ViewHitCounterViewerEndpointDeployment1CE7C576aaeb52ee3dd5ec3ab243f366d5eae8a9",
        },
        "RestApiId": {
          "Ref": "ViewHitCounterViewerEndpoint5A0EF326",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "ViewHitCounterViewerEndpointproxy2F4C239F": {
      "Properties": {
        "ParentId": {
          "Fn::GetAtt": [
            "ViewHitCounterViewerEndpoint5A0EF326",
            "RootResourceId",
          ],
        },
        "PathPart": "{proxy+}",
        "RestApiId": {
          "Ref": "ViewHitCounterViewerEndpoint5A0EF326",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "ViewHitCounterViewerEndpointproxyANYApiPermissionMainStackViewHitCounterViewerEndpoint90EA86D9ANYproxy8B7D9C4D": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "ViewHitCounterRendered9C783E45",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "ViewHitCounterViewerEndpoint5A0EF326",
              },
              "/",
              {
                "Ref": "ViewHitCounterViewerEndpointDeploymentStageprodF3901FC7",
              },
              "/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ViewHitCounterViewerEndpointproxyANYApiPermissionTestMainStackViewHitCounterViewerEndpoint90EA86D9ANYproxy33E4DB92": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "ViewHitCounterRendered9C783E45",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "ViewHitCounterViewerEndpoint5A0EF326",
              },
              "/test-invoke-stage/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ViewHitCounterViewerEndpointproxyANYFF4B8F5B": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "ViewHitCounterRendered9C783E45",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "ViewHitCounterViewerEndpointproxy2F4C239F",
        },
        "RestApiId": {
          "Ref": "ViewHitCounterViewerEndpoint5A0EF326",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
